x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/rest/action/RestClearSecurityCacheAction.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+package org.elasticsearch.xpack.security.rest.action;
+
+import org.elasticsearch.client.node.NodeClient;
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.license.XPackLicenseState;
+import org.elasticsearch.rest.RestRequest;
+import org.elasticsearch.rest.action.RestActions.NodesResponseRestListener;
+import org.elasticsearch.xpack.core.security.action.ClearSecurityCacheAction;
+import org.elasticsearch.xpack.core.security.action.ClearSecurityCacheRequest;
+
+import java.io.IOException;
+import java.util.Collections;
+import java.util.List;
+
+import static org.elasticsearch.rest.RestRequest.Method.POST;
+
+public class RestClearSecurityCacheAction extends SecurityBaseRestHandler {
+
+    public RestClearSecurityCacheAction(Settings settings, XPackLicenseState licenseState) {
+        super(settings, licenseState);
+    }
+
+    @Override
+    public String getName() {
+        return "security_clear_cache_action";
+    }
+
+    @Override
+    public List<Route> routes() {
+        return Collections.singletonList(new Route(POST, "/_security/{cache_name}/{keys}/_clear_cache"));
```

Comment:
Sorry this is a mistake on my side and thanks for catching it. My intention for the new Clear Cache API is just an internal refactor and should not require user awareness. This is why I kept the Rest end point to be compatible with the existing ones, e.g. `/_security/privilege/{application}/_clear_cache`, where `privilege` is the cache name and `{application}` is the keys. My plan was to introduce the new cache clearing API as if it is just for API key, i.e, I planned to document it just like all the other existing APIs and what user should be seeing is just `/_security/api_key/{id}/_clear_cache`.

But somehow I got carried away by the changes and added documentation for the generic API itself, which like you said, is low level and not user friendly. Hence I will drop this documentation and just document it in the context of API key. Whether these cache clearing APIs internally use a single or different actions are implementation details and should not concern users.

Reply:
> On the surface (documentation), user will just see that we add another cache clearing API for API keys, i.e. /_security/api_key/{id}/_clear_cache.

The `_security/api_key` APIs have the parameters in the request body because of the multiple ways to refer to keys (by `id`, `name`, `user` and `realm`). If you put in the `id` in the URL path for the cache clear operation, it would be unlike the other APIs and would be more complex to introduce clearing of the cache for keys by other params.

Reply:
I don't think we want to match the cache clearing API with other "api_key" APIs. It is never a goal to clear cache by API key names, users or realms. Caches are always cleared by the keys, i.e. `id` in API key's case.  So I don't see any problem here.

Reply:
@albertzaharovits Per our discussion, I have [changed](https://github.com/elastic/elasticsearch/pull/59376/commits/11e591babca2733bc1ce37c692989e38c9586350#diff-f0edbeeb832bf03f92fffc111d46fbd2) the Rest Action class to be dedicated to "clear API key cache". Thanks for the feedback!